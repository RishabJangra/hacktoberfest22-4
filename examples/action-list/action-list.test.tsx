import React from 'react';
import renderer from 'react-test-renderer';
import { ActionListScreen, ListItem } from './ActionList';
import Modal from 'react-native-modal';
import { InfoListItem } from '@pxblue/react-native-components';
import { FlatList } from 'react-native';
import { render, fireEvent } from '@testing-library/react-native';

jest.mock('@react-navigation/native', () => ({
    useNavigation: (): any => ({ openDrawer: jest.fn(() => true) }),
}));

const hardcodedData: ListItem[] = [
    {
        id: 1,
        name: 'item 1',
        details: 'item 1 details'
    },
    {
        id: 2,
        name: 'item 2',
        details: 'item 2 details'
    },
    {
        id: 3,
        name: 'item 3',
        details: 'item 3 details'
    }
]

describe('Action List Tests', () => {
    it('Screen Renders', () => {
        const tree = renderer.create(<ActionListScreen hardcodedData={hardcodedData} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Should render 3 items if 3 items are passed in', () => {
        const instance = renderer.create(<ActionListScreen hardcodedData={hardcodedData} />).root;
        const flatList = instance.findByType(FlatList);
        const infoListItems = flatList.findAllByType(InfoListItem);
        expect(infoListItems).toHaveLength(3);
    });

    it('Should render 10 items by default', () => {
        const instance = renderer.create(<ActionListScreen />).root;
        const flatList = instance.findByType(FlatList);
        const infoListItems = flatList.findAllByType(InfoListItem);
        expect(infoListItems).toHaveLength(10);
    });

    xit('Should add an Item from empty state', () => {
        const emptyData: ListItem[] = []
        const instance = render(<ActionListScreen hardcodedData={emptyData}/>);
        const addButton = instance.getByTestId("empty-state-add-button");
        fireEvent.press(addButton);

        const flatList = instance.getByTestId("list");
        const infoListItems = flatList.findByType(InfoListItem);
        expect(infoListItems).toHaveLength(1);
    });

    xit('Should add an Item', () => {
        const instance = render(<ActionListScreen />);
        const header = instance.getByTestId("header");
        const addButton = header.findAllByType("button")[2]; // just have to find the button... ahhhh!!!
        fireEvent.press(addButton)

        const flatList = instance.getByType(FlatList);
        const infoListItems = flatList.findAllByType(InfoListItem);
        expect(infoListItems).toHaveLength(11);


        // const instance = renderer.create(<ActionListScreen />).root;
        // instance.props.addItem();
        // const addButton = renderer.create(<Button onPress={addItem} accessibilityStates="test">Click Me</Button>)

        // const header = instance.findByType(Header);
        // header.props.actionItems[1].handleClick();
        // const flatList = instance.findByType(FlatList);
        // const infoListItems = flatList.findAllByType(InfoListItem);
        // expect(infoListItems).toHaveLength(11);
        
        // handleClick();
        // handleClick();
        // handleClick();
        // expect(infoListItems).toHaveLength(14);
    });

    // xit('Delete an Item', () => {
    //     const app = shallow(<ActionListScreen />);
    //     app.instance().setSelectedItemIndex(2);
    //     app.instance().onDelete();
    //     expect(app.data.length).toBe(9);
    //     app.instance().setSelectedItemIndex(7);
    //     app.instance().onDelete();
    //     app.instance().setSelectedItemIndex(4);
    //     app.instance().onDelete();
    //     expect(app.data.length).toBe(7);
    // });

    // xit('Delete all Items', () => {
    //     const app = shallow(<ActionListScreen />);
    //     app.instance().deleteAll();
    //     expect(app.data.length).toBe(0);
    // });

    // xit('Modal hidden by default', () => {
    //     const app = shallow(<ActionListScreen />);
    //     expect(app.find(Modal).props().isVisible).toBe(false);
    // })

    // xit('Show modal when clicking the menu', () => {
    //     const app = shallow(<ActionListScreen />);
    //     app.instance().showModal(2);
    //     expect(app.find(Modal).props().isVisible).toBe(true);
    // })
});
