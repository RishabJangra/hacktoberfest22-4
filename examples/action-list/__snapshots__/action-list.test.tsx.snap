// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action List Tests Screen Renders 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    accessible={true}
    emulateUnlessSupported={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#6200ee",
        "elevation": 0,
        "height": 76,
        "shadowColor": "rgba(0, 0, 0, 0.3)",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 2,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "paddingBottom": 16,
          "paddingHorizontal": 16,
          "paddingTop": 16,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 40,
            "margin": -8,
            "marginRight": 24,
            "opacity": 1,
            "padding": 8,
            "width": 40,
          }
        }
        testID="header-navigation"
      >
        <Text />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-end",
            "marginRight": 80,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 0,
              "justifyContent": "center",
            }
          }
        >
          <Text
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "white",
                "fontFamily": "System",
                "fontSize": 20,
                "lineHeight": 20,
              }
            }
            testID="header-title"
          >
            Action List
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 56,
              "position": "absolute",
              "right": 8,
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": 40,
              "opacity": 1,
              "padding": 8,
              "width": 40,
            }
          }
          testID="header-action-item0"
        >
          <Text />
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": 40,
              "opacity": 1,
              "padding": 8,
              "width": 40,
            }
          }
          testID="header-action-item1"
        >
          <Text />
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
  <RCTScrollView
    data={
      Array [
        Object {
          "details": "item 1 details",
          "id": 1,
          "name": "item 1",
        },
        Object {
          "details": "item 2 details",
          "id": 2,
          "name": "item 2",
        },
        Object {
          "details": "item 3 details",
          "id": 3,
          "name": "item 3",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    testID="list"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "flexDirection": "row",
              "height": 72,
              "opacity": 1,
              "paddingRight": 16,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": undefined,
                  "height": "100%",
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 6,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "paddingHorizontal": 16,
                },
                undefined,
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    undefined,
                  ],
                ]
              }
            >
              item 1
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0,
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    undefined,
                    Object {
                      "color": "#000000",
                    },
                  ]
                }
              >
                item 1 details
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <Text />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "flexDirection": "row",
              "height": 72,
              "opacity": 1,
              "paddingRight": 16,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": undefined,
                  "height": "100%",
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 6,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "paddingHorizontal": 16,
                },
                undefined,
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    undefined,
                  ],
                ]
              }
            >
              item 2
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0,
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    undefined,
                    Object {
                      "color": "#000000",
                    },
                  ]
                }
              >
                item 2 details
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <Text />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "flexDirection": "row",
              "height": 72,
              "opacity": 1,
              "paddingRight": 16,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": undefined,
                  "height": "100%",
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 6,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "paddingHorizontal": 16,
                },
                undefined,
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    undefined,
                  ],
                ]
              }
            >
              item 3
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0,
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    undefined,
                    Object {
                      "color": "#000000",
                    },
                  ]
                }
              >
                item 3 details
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <Text />
        </View>
      </View>
    </View>
  </RCTScrollView>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  >
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      onRequestClose={[Function]}
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
      transparent={true}
      visible={false}
    >
      <View
        accessible={true}
        focusable={true}
        forwardedRef={[Function]}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "bottom": 0,
            "height": 1334,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 750,
          }
        }
      />
      <View
        forwardedRef={[Function]}
        hideModalContentWhileAnimating={false}
        onModalHide={[Function]}
        onModalWillHide={[Function]}
        onModalWillShow={[Function]}
        pointerEvents="box-none"
        scrollHorizontal={false}
        scrollOffset={0}
        scrollOffsetMax={0}
        scrollTo={null}
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
            "margin": 0,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        swipeThreshold={100}
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "flexDirection": "row",
                "height": 72,
                "opacity": 1,
                "paddingRight": 16,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": undefined,
                    "height": "100%",
                    "left": 0,
                    "position": "absolute",
                    "top": 0,
                    "width": 6,
                  },
                  undefined,
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-start",
                    "justifyContent": "center",
                    "marginLeft": 16,
                    "width": 40,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={null}
              >
                <Text />
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "paddingHorizontal": 16,
                  },
                  undefined,
                ]
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                    Array [
                      Object {
                        "color": "#000000",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                Remove
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    },
                    undefined,
                  ]
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    },
                    undefined,
                  ]
                }
              />
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "flexDirection": "row",
                "height": 72,
                "opacity": 1,
                "paddingRight": 16,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": undefined,
                    "height": "100%",
                    "left": 0,
                    "position": "absolute",
                    "top": 0,
                    "width": 6,
                  },
                  undefined,
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-start",
                    "justifyContent": "center",
                    "marginLeft": 16,
                    "width": 40,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={null}
              >
                <Text />
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "paddingHorizontal": 16,
                  },
                  undefined,
                ]
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                    Array [
                      Object {
                        "color": "#000000",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                Cancel
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    },
                    undefined,
                  ]
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    },
                    undefined,
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
    </Modal>
  </RCTSafeAreaView>
</View>
`;
